spring-cloud-gateway-server/src/main/java/org/springframework/cloud/gateway/route/builder/RouteLocatorBuilder.java

```diff
@@ -98,7 +101,19 @@ ConfigurableApplicationContext getContext() {
 		}
 
 		void add(Route.AsyncBuilder route) {
-			routes.add(route);
+			if (route.isEnableDefaultFilters()) {
+				GatewayProperties gatewayProperties = context
+						.getBean(GatewayProperties.class);
+				RouteDefinitionRouteLocator routeDefinitionRouteLocator = context
```

Comment:
I'd prefer an abstraction, maybe `DefaultRoutes` with a `getDefaultGatewayFilters()` method as to not introduce tangle.